### ISSUE TITLE
Wrap account public keys in protos and upload to XMTP topics

### ISSUE THREAD

**Issue by m-005** [xmtp/xmtp-js#347]
Wrap the account public keys into protos in rust-land, and upload them (likely from JS)
Proto generation is handled by xmtp-proto crate

Pre-requisites:
https://github.com/xmtp/proto/issues/81
https://github.com/xmtp/libxmtp/issues/73

**Comment by michaelx11** [comment]
This task also depends on #349 right? We need networking to upload to XMTP topics. I can get started on a Rust-land networking backend + a networking trait.

**Comment by michaelx11** [comment]
releasing this task

**Comment by michaelx11** [comment]
I've released this task back into the pool. I'll give a quick summary of my thoughts here, although I didn't really get to dig into it.

I merged a PR last week that wraps contact elements in protos: https://github.com/xmtp/libxmtp/pull/89

To upload to XMTP topics we can:
1) Choose the topic naming structure (could just append a new prefix and switch to `/xmtp/1/` per the MudPup demo [sprint](https://github.com/xmtp/xmtp-js/blob/vodozemac_dev/src/voodoo/utils.ts#L3).
2) Publish to the topic. This can happen in one of two ways:
  - Via xmtp-js - can either try to consume it in `bindings_js` or have a branch in `xmtp-js` that consumes our wasm bundle. The latter requires running tests from xmtp-js.
  - Via Rust (xmtp_networking). xmtp_networking is well understood but does _not_ work as it is for wasm land (`tonic` dependency doesn't build for wasm). We can still utilize it for unit tests if we keep enforce a well-defined `Networking` trait.

Either way, the first step may be to just define that `Networking` trait and mock it out. Then we can build various backends for it.
